server/index.js

```diff
@@ -86,6 +88,42 @@ app.get('/', function(req, res){
     .catch(function(err){ console.error('/', err, err.stack); }); 
 });
 
+app.get('/place/:placeId/', function(req, res){
+
+    console.log('==== calling /place/:placeId/')
+    // Create a fresh document every time
+    makeDocument(indexHTMLStr).then(function(result){
+        var doc = result.document;
+        var dispose = result.dispose;
+
+        var placeId = Number(req.params.placeId);
+        places.getPlaceById(placeId)
+        .then(function(data){
+            toGeoJson(data)
+            .then(function(geoJson){
+            
+                layoutData.detailedObject = geoJson[0];
+
+                // Material-ui needs to change the global.navigator.userAgent property
+                // (just during the React  rendering)
+                // Waiting for Material team to fix it 
+                //https://github.com/callemall/material-ui/pull/2007#issuecomment-155414926
+                global.navigator = {'userAgent': req.headers['user-agent']};
+                renderDocumentWithData(doc, layoutData, Layout);
```

Comment:
A first problem we should be taking care of is that both `/` and `/place/:placeId` are rendered using the same React component (`Layout`) although they should render something different.

`Layout` most renders something different based on the props, but this is not really scalable and/or maintainable.
The approach I've been using in MyWI is to have as many top-level components as I have routes and prefix them with `-Screen`. See https://github.com/MyWebIntelligence/MyWebIntelligence/tree/master/client/components (`OraclesScreen`, `TerritoireListScreen`, `TerritoireViewScreen`). This way, it's crystal-clear that there is a bijection ( ;-) ) between routes and React "screen components" and the bug we're encountering right now (mismatch between the URL and what's being displayed) should be impossible by design.

Obviously, inside these top-level components, the goal is to reuse as many smaller components as possible (headers, etc.).

In our case, we probably should be having something like `MainScreen` (for `/`) and `PlaceScreen` (for `/place/:placeId`)